### DISCUSSION TITLE
zipkin exporter encoding

### DISCUSSION THREAD

**Discussion by jkwatson** [open-telemetry/opentelemetry-java#2556] [Q&A]
Currently, the zipkin exporter defaults to encoding as `json`. If you want to use the protobuf encoding, you have to manually configure the exporter like this:

```java
        ZipkinSpanExporter.builder()
            .setServiceName(SERVICE_NAME)
            .setEncoder(SpanBytesEncoder.PROTO3)
            .setSender(OkHttpSender.newBuilder()
                .endpoint(httpUrl + ENDPOINT_V2_SPANS)
                .encoding(Encoding.PROTO3)
                .build())
            .build();
```

This is pretty non-intuitive, and you definitely have to know what you're doing to get this to work. 

2 questions:

1) should we keep json as the default encoding for the zipkin exporter
2) should we make it easier to configure the encoding, rather than make users know the guts of how to make it work?

**Comment by anuraaga** [comment]
1. protobuf default sounds good
2. Yeah adding a method for setting format looks worth it
